### DISCUSSION TITLE
Issue with app deployment

### DISCUSSION THREAD

**Discussion by mcnold** [community/community#45359] [New to GitHub]
Hi there! I am not sure if this is the correct area to post this issue, but I figured I would start here. I am attempting to make an update to my app repo. It appears to have successfully deployed, but I am getting the following error from GitHub's side:

<img width="960" alt="image" src="https://user-images.githubusercontent.com/80659979/214635018-b99fca1d-10f2-474e-9ca5-b3a681b64b3d.png">

It gives me the error message of "pre-received hook declined." I tried searching the repo for previous examples, but it says to contact GitHub to resolve. 

Thanks for any advice!

**Comment by jge162** [comment]
@mcnold The error message "pre-received hook declined" typically indicates that a pre-receive hook on the GitHub repository has rejected the push 

you'll need to check the pre-receive hooks that are configured on the repository to see which one is rejecting the push. You can check the pre-receive hooks by going to the repository's settings, then the "Hooks" tab. You can also check the logs of the hook that is rejecting the push to see why it's failing.

If you can't find the problem by looking at the hooks or logs, you can also try disabling the hooks temporarily to see if the push goes through.

**Reply by mcnold** [reply]
Thank you, I will try those suggestions!

**Reply by jge162** [reply]
@mcnold It's also worth mentioning that, if you are using GitHub Actions, it could be that an action is failing and that's why the hook is declining your push. Make sure to check the logs of the actions to see if they are passing or failing. Just Incase to be sure. Hope you find issue.

**Reply by mcnold** [reply]
Thank you. Yes, i deleted the hook and it's still occurring. I will check actions.

**Reply by mcnold** [reply]
I don't see anything...

**Reply by jge162** [reply]
Maybe the local repository is out of sync with the remote repository.


git fetch origin

git status

git commit -am "message"

git push origin branch_name


If  you are not using any actions and you've checked the logs and there's nothing, you could try another approach, you can try to push the changes to a new branch and then open a pull request for that branch, this way you are not modifying the main branch

**Reply by mcnold** [reply]
Okay thank you I will try that

**Reply by mcnold** [reply]
It says everything is up to date. I am on the main branch (I forked the repo and cloned it originally)

**Reply by jge162** [reply]
@mcnold Make sure that you have write access to the repository. If you forked the repository, you will only have write access to your fork, not the original repository.

Check if there are any branch protection rules that are blocking the push. Go to the repository's settings, then the "Branches" tab, to see if there are any rules that are preventing the push.

check if the repository has been set up to require pull requests. You can check this by going to the repository's settings, then the "Collaborators & teams" tab.

check if the repository has been set up to use a template. You can check this by going to the repository's settings, then the "Options" tab.

that’s all I got. Hopefully it works. You might need to reach out to GitHub support.

**Reply by mcnold** [reply]
Thank you, no the original contributor allowed forking. I appreciate all your suggestions though. Where do I contact support?

**Reply by jge162** [reply]
https://support.github.com/request?q=Hook+error
you can start here. Just choose the appropriate issue or “other” at the bottom. Or can submit bug report too. Sorry l this was not resolved.  Best of luck to you.

**Reply by mcnold** [reply]
Thank you, no I appreciate the multiple tries and suggestions, truly. Enjoy your week!
